engine/commands/cortex_upd_cmd.cc

```diff
@@ -61,69 +64,83 @@ bool CortexUpdCmd::GetStableAndBeta(const std::string& v) {
-            std::string path = full_url.substr(host.length());
-
-            auto fileName = asset["name"].get<std::string>();
-            CTL_INF("URL: " << full_url);
-
-            auto download_task = DownloadTask{.id = "cortex",
-                                              .type = DownloadType::Cortex,
-                                              .error = std::nullopt,
-                                              .items = {DownloadItem{
-                                                  .id = "cortex",
-                                                  .host = host,
-                                                  .fileName = fileName,
-                                                  .type = DownloadType::Cortex,
-                                                  .path = path,
-                                              }}};
-
-            DownloadService download_service;
-            download_service.AddDownloadTask(
-                download_task,
-                [this](const std::string& absolute_path, bool unused) {
-                  // try to unzip the downloaded file
-                  std::filesystem::path download_path{absolute_path};
-                  CTL_INF("Downloaded engine path: " << download_path.string());
-
-                  std::filesystem::path extract_path =
-                      download_path.parent_path().parent_path();
-
-                  archive_utils::ExtractArchive(download_path.string(),
-                                                extract_path.string());
-
-                  CTL_INF("Finished!");
-                });
+      } catch (const nlohmann::json::parse_error& e) {
+        CTL_ERR("JSON parse error: " << e.what());
+        return false;
+      }
+    } else {
+      CTL_ERR("HTTP error: " << res->status);
+      return false;
+    }
+  } else {
+    auto err = res.error();
+    CTL_ERR("HTTP error: " << httplib::to_string(err));
+    return false;
+  }
+
+  // Replace binary file
+  auto executable_path = file_manager_utils::GetExecutableFolderContainerPath();
+  auto src = std::filesystem::temp_directory_path() / "cortex" / kCortexBinary /
+             GetCortexBinary();
+  auto dst = executable_path / GetCortexBinary();
+  return ReplaceBinaryInflight(src, dst);
+}
+
+bool CortexUpdCmd::GetBeta(const std::string& v) {
+  // Check if the architecture and OS are supported
+  auto system_info = system_info_utils::GetSystemInfo();
+  if (system_info.arch == system_info_utils::kUnsupported ||
+      system_info.os == system_info_utils::kUnsupported) {
+    CTL_ERR("Unsupported OS or architecture: " << system_info.os << ", "
+                                               << system_info.arch);
+    return false;
+  }
+  CTL_INF("OS: " << system_info.os << ", Arch: " << system_info.arch);
+
+  // Download file
+  constexpr auto github_host = "https://api.github.com";
+  std::ostringstream release_path;
+  release_path << "/repos/janhq/cortex.cpp/releases";
```

Comment:
@vansangpfiev Just a note, we may need to support multiple download sources in the future (e.g. Github, and mirrors in Gitlab, Gitee)